src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement2.java

```diff
@@ -59,29 +59,29 @@ public void checkBooleanStatements() {
                         && (a // violation 'Unnecessary parentheses around expression'
                         && (b==c))) // violation 'Unnecessary parentheses around expression'
                 || (a // violation 'Unnecessary parentheses around expression'
-                || a!=(b<=c)) // ok
-                || (a==(b!=d==(c==b) && a!=(b<=c)))) { // violation 'parentheses around expression'
+                || a!=b<=c) // ok
+                || (a==(b!=d==(c==b) && a!=b<=c))) { // violation 'parentheses around expression'
                                                        // after '||'
```

Comment:
@nrmancuso All the implemenattion looks correct to me new report has no major difference then previous one.

All the new changes in new report then previous one are something similar to the code for i am leaving this comment in and they all are correct as per rule of operator precedence. we can remove this braces.

In the previous implementation, it is been not considered.  But know as per the new implementation we directly check the precedence of operators so i guess this may be bug which resolved by new implemnattion

Reply:
@Kevin222004 thanks a lot for finding this. You are correct, this should be a violation. However, I think this many changes in behavior in one PR/issue might be too crazy. I propose the following:
1. Let's go back to https://github.com/checkstyle/checkstyle/commit/925e6cf8f2acbbee30959031578ca10ef9c7f0f7, since we add only new functionality here
2. Let's open an issue for what you found above, and we can address this later.

From what I understand, this check has been controversial in the past for being too strict, so we should be careful in changing behavior too much. As you return to old commit, please update this documentation with a couple examples of new tokens: https://checkstyle.org/config_coding.html#UnnecessaryParentheses:~:text=The%20partial%20support%20for%20operator%20precedence%20includes%20cases%20of%20the%20following%20type%3A

Reply:
@nrmancuso i update the code as previous implementation but

 if i open the new issue regarding the new code changes then, then lets say if we expand the code as in https://github.com/checkstyle/checkstyle/commit/925e6cf8f2acbbee30959031578ca10ef9c7f0f7 then it might be very complex to understand.
 
I dont think their would be any other better way to implement this as it is right know 

>From what I understand, this check has been controversial in the past for being too strict, so we should be careful in changing behavior too much

I think we can make issue some how for selected operator like Bitwise for that we can remove parentheseis if more then 1 is their. because normally the presence of  operator like bitwise  is difficult to remember and also to having parantehsis is a good way rather than bad to understand code simple [please look] (https://github.com/CS-SI/Orekit/pull/32#:~:text=I%20understand%20it%20follows%20Java%20well%2Ddefined%20operator%20priority%20semantics%2C%20but%20these%20semantics%20are%20sometimes%20hard%20to%20remember%20for%20mere%20humans.%0AI%20usually%20use%20parentheses%20as%20a%20kind%20of%20in%2Dcode%20help%20to%20ease%20maintenance%2C%20so%20for%20me%2C%20these%20extra%20parentheses%20are%20rather%20good%20style%20than%20bad%20style.)

Reply:
https://github.com/checkstyle/checkstyle/pull/12703#discussion_r1158481390 done

Reply:
@nrmancuso ping